### DISCUSSION TITLE
Addin directive errors with non-standard nuget packages.

### DISCUSSION THREAD

**Discussion by phillipsj** [cake-build/cake#3177] [Q&A]
When trying to using NuGet packages that don't use the "lib/net45" content location with the AddIn directive, I am receiving the following error:

    Could not find any assemblies compatible with .NETFramework, Version=v4.5.
    Error:  Failed to install addin 'WinSCP'

How to recreate:

1.  Add the following AddIn directive

    #tool nuget:?package=WinSCP

2.  Run build.ps1 and receive the error.

![addin-error](https://cloud.githubusercontent.com/assets/1568133/23465975/ce4846a4-fe67-11e6-886c-59959e85ee8d.png)
![addin-error2](https://cloud.githubusercontent.com/assets/1568133/23465980/d192daa4-fe67-11e6-9453-1e89d6eb2750.png)
![addin-error1](https://cloud.githubusercontent.com/assets/1568133/23465982/d389bf12-fe67-11e6-83a0-47868ee1d3eb.png)

There are several libraries that do not use the standards, this is just one example.

**Comment by phillipsj** [comment]
The work around is to pull it down as a tool and use the reference directive to pull it in.

**Comment by patriksvensson** [comment]
@phillipsj In this case your only option is using `#tool` with `#r` like you mention that you do. You will encounter the same problem in VS. 

Unless the WinScp package author changes the package to conform to the standard NuGet conventions it's not much we could do about it at the moment :/

What we theoretically could do in the future, that might be usable for other packages, is to add a parameter to explicitly include files in a NuGet package.

```csharp
#addin nuget:?package=Foo&ref=bar/baz.dll&ref=qux/bar.dll
```

**Comment by phillipsj** [comment]
@patriksvensson Now that sounds like a cool feature, that way users could "opt-in" to load those if you know they can use it.

**Comment by devlead** [comment]
Though, preferably with .NET Core hopefully soon going mainstream, ideally addins should use correct framework monikers.

**Comment by phillipsj** [comment]
@devlead @patriksvensson Cool, you all can close this if you want, its fine with me.
